mllib/src/main/scala/org/apache/spark/ml/feature/LSH.scala

```diff
@@ -179,16 +211,13 @@ private[ml] abstract class LSHModel[T <: LSHModel[T]]
       inputName: String,
       explodeCols: Seq[String]): Dataset[_] = {
     require(explodeCols.size == 2, "explodeCols must be two strings.")
-    val vectorToMap = udf((x: Vector) => x.asBreeze.iterator.toMap,
-      MapType(DataTypes.IntegerType, DataTypes.DoubleType))
     val modelDataset: DataFrame = if (!dataset.columns.contains($(outputCol))) {
       transform(dataset)
     } else {
       dataset.toDF()
     }
     modelDataset.select(
-      struct(col("*")).as(inputName),
-      explode(vectorToMap(col($(outputCol)))).as(explodeCols))
+      struct(col("*")).as(inputName), posexplode(col($(outputCol))).as(explodeCols))
```

Comment:
If you look at line 292 of `Executor.scala`, it shows this is just a OOM exception of Dataframe. That's the reason why it behaves different on our machines and Jenkins.
`model.approxSimilarityJoin(df, df, Double.MaxValue, "distCol")` returns near 40000 rows when `threshold = Double.MaxValue`
If you reduce the `numDim`, the test will pass.